expression/builtin_time_vec.go

```diff
@@ -1467,11 +1467,43 @@ func (b *builtinAddDateIntRealSig) vecEvalTime(input *chunk.Chunk, result *chunk
 }
 
 func (b *builtinSubDurationAndDurationSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinSubDurationAndDurationSig) vecEvalDuration(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	if err := b.args[0].VecEvalDuration(b.ctx, input, result); err != nil {
+		return err
+	}
+	buf0 := result
+	buf1, err := b.bufAllocator.get(types.ETDuration, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf1)
+	if err := b.args[1].VecEvalDuration(b.ctx, input, buf1); err != nil {
+		return err
+	}
+	result.MergeNulls(buf1)
+	arg0s := buf0.GoDurations()
+	arg1s := buf1.GoDurations()
+	resultSlice := result.GoDurations()
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		// get arg0 & arg1
+		arg0 := arg0s[i]
+		arg1 := arg1s[i]
+		// calculate
+		output, err := types.AddDuration(arg0, -arg1)
```

Comment:
Will this cause a overflow? `arg1` is a `uint64` . I prefer to use `SubUint64` here.

Reply:
![image](https://user-images.githubusercontent.com/22973642/69524319-b12e3180-0fa0-11ea-8ce0-96788d846e24.png)
But you can see in this AddDuration function,Isn't it already guaranteed that it will not overflow?

Reply:
Sorry, i will change it....

Reply:
I have change it now. But  I think arg1s[i] is int64... So I use Subint64 here.
I still have a question:Why the subtraction will cause overflow.

Reply:
I thought duration is a `uint64` before, sorry for the misleading.